src/Microsoft.Graph.Core/Authentication/TokenCredentials/IntegratedWindowsTokenCredential.cs

```diff
@@ -0,0 +1,155 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph
+{
+    using Azure.Core;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using System.Collections.Generic;
+    using Microsoft.Identity.Client;
+    using System;
+
+    /// <summary>
+    /// A <see cref="TokenCredential"/> implementation using MSAL.Net to acquire token by integrated windows authentication.
+    /// </summary>
+    public class IntegratedWindowsTokenCredential : TokenCredential
+    {
+
+        private static readonly List<string> WellKnownTenants = new List<string>
+        {   AuthConstants.Tenants.Common,
+            AuthConstants.Tenants.Consumers,
+            AuthConstants.Tenants.Organizations
+        };
+
+        private readonly IPublicClientApplication _clientApplication;
+
+        /// <summary>
+        /// Creates a new IntegratedWindowsTokenCredential which will authenticate users with the specified application.
+        /// </summary>
+        /// <param name="publicClientApplication">A <see cref="IPublicClientApplication"/> to pass to <see cref="IntegratedWindowsTokenCredential"/> for authentication.</param>
+        public IntegratedWindowsTokenCredential(IPublicClientApplication publicClientApplication)
+        {
+            _clientApplication = publicClientApplication ?? throw new AuthenticationException(
```

Comment:
We should throw a ArgumentNullException if publicClientApplication is null. We haven't attempted to authenticate yet at this point. Also, we should document `///<exception></exception>` thrown exceptions